identity_credential/src/revocation/status_list_2021/entry.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020-2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_core::common::Url;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::credential::Status;
+
+use super::credential::StatusPurpose;
+
+const CREDENTIAL_STATUS_TYPE: &str = "StatusList2021Entry";
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash, Serialize)]
+#[serde(transparent)]
+struct EntryType(&'static str);
```

Comment:
It could have been easier to use serde's `deserialize_with` and check the string equality there, instead of introducing this struct. That would be much more readable IMO.

Reply:
If I want to avoid `EntryType` I'll need a `&'static str type_` field in `StatusList2021Entry`. To deserialize that I can use a function with the signature:
`fn<'de, D: Deserializer>(d: D) -> Result<&'static, D::Error>`. It will work but there's an issue: `StatusList2021Entry` implements `Deserialize<'static>` instead of using any lifetime, this makes it impossible to deserialize for any `&str` other than `&'static str`, which is a no go.
Going the `EntryType` way solves this, another solution could be to simply clone and use `String` instead of borrowing from the deserializer.  Lemme know what do you think the best approach would be.